### DISCUSSION TITLE
3.0.0-rc.11 Configure 404 error page

### DISCUSSION THREAD

**Discussion by echo9z** [nuxt/nuxt#16720] [Questions]
Add error.vue to ~/, request a routing address that does not exist, or the default nuxt3 error page.
package.json
```
  "devDependencies": {
    "@nuxtjs/tailwindcss": "^5.3.3",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "@vueuse/core": "^9.2.0",
    "@vueuse/nuxt": "^9.2.0",
    "eslint": "^8.0.1",
    "eslint-config-standard-with-typescript": "^23.0.0",
    "eslint-plugin-import": "^2.25.2",
    "eslint-plugin-n": "^15.0.0",
    "eslint-plugin-promise": "^6.0.0",
    "eslint-plugin-vue": "^9.5.1",
    "nuxt": "3.0.0-rc.11",
    "sass": "^1.55.0",
    "typescript": "*"
  },
  "dependencies": {
    "@pinia/nuxt": "^0.4.2"
  }
```

**Comment by sevillaarvin** [comment]
Can you clarify the question?

**Reply by echo9z** [reply]
This problem also existed before using rc-10，eg：https://github.com/echo9z/study_nuxt3

**Reply by sevillaarvin** [reply]
You need to install `pinia` with `@pinia/nuxt`

    "@pinia/nuxt": "^0.4.2",
    "pinia": "^2.0.20"

**Reply by echo9z** [reply]
Tried it, nothing to do with pinia

**Comment by bot08** [comment]
If you want custom 404 page, just create a 404.vue file in the pages directory. If it's a dynamic url, use createerror.

**Comment by dpmango** [comment]
@bot08 `~/pages/404.vue` will work. 

Alternative option is to use `~/error.vue` (root folder) with `error prop`

P.S. Would be nice to move this somewhere on the root using `nuxt.config`

```vue
<template lang="pug">
.page
  LayoutHeader
  LayoutError404(v-if="error.statusCode === '404'")
  LayoutError500(v-else)
</template>

<script setup>
const props = defineProps({
  error: Object,
})

useHead({
  title: `Error ${props.error.statusCode}`,
})
</script>
```

**Reply by sevillaarvin** [reply]
Did you place it under `~/pages/404.vue`?

https://v3.nuxtjs.org/guide/directory-structure/pages#catch-all-route

> Nuxt also supports a custom pages/404.vue file which will handle all routes that are not otherwise matched (and set a 404 error code).

<img width="1086" alt="image" src="https://user-images.githubusercontent.com/33614172/192939665-2b48205d-2373-4574-8973-95fafa55c17d.png">

**Reply by echo9z** [reply]
Thank. I just tried. Just add 404.vue to/pages/404.vue

**Reply by bot08** [reply]
@dpmango why u need this overengineering...

**Reply by dpmango** [reply]
pages/404.vue works, initially thought about the root path 

sorry for confusion
